Fix: sort-keys in an object that contains spread (fixes #10261)

katerberg opened the pull request:
<!--
    ESLint adheres to the [JS Foundation Code of Conduct](https://js.foundation/community/code-of-conduct).
-->

**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[X] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

<!--
    If the item you've checked above has a template, please paste the template questions below and answer them. (If this pull request is addressing an issue, you can just paste a link to the issue here instead.)
-->

<!--
    Please ensure your pull request is ready:

    - Read the pull request guide (https://eslint.org/docs/developer-guide/contributing/pull-requests)
    - Include tests for this change
    - Update documentation for this change (if appropriate)
-->

<!--
    The following is required for all pull requests:
-->

**What changes did you make? (Give an overview)**
I caused the linter to ignore any key-ordering in an object that contains a spread operator. Previously, it would verify that any keys that were not themselves spread operators were in the correct order, but this caused issues when the spread operator was between other keys as illustrated in https://eslint.org/docs/developer-guide/contributing/pull-requests

**Is there anything you'd like reviewers to focus on?**
This is my first pull request to this project, so any feedback would be welcomed. Since this does remove a test, I would also like to have a second set of eyes verify that the original test was indeed accidental and should have been reversed.

platinumazure commented:
@katerberg Friendly ping... Did you have any questions about my review, and/or is there anything else I can do to help? Thanks!

ljharb commented:
Hmm, why would we ignore all sorting? I'd think that the keys would be split up into groups - with a spread as the boundary - and each group would be individually sorted.

katerberg commented:
@platinumazure I'm still going to do this, but have had a busy weekend. Hoping to get to it next weekend.

@ljharb That would be reasonable as well. Do you and @platinumazure agree that it should be the behavior? If so, I can change the PR to match those goals.

platinumazure commented:
@katerberg I'm okay with trying to report on sorting keys in the "groups" before and after the object rest property node, if that seems feasible to you.

I can't remember if the rule currently does autofix or not, but if it does, I'll note that rules do not *always* have to autofix if they autofix in other cases: Users know that autofix is not guaranteed. So basically, I'm suggesting not to bite off too much with this bugfix. Incremental improvements are totally okay here. Thanks again for contributing!

katerberg commented:
Sounds good. I'll likely keep it as it is for now and make an issue to add the grouped alphabetizing as a separate PR. It looks like autofix is off for this rule on purpose: https://github.com/eslint/eslint/issues/10324

katerberg commented:
I've now added the documentation to make the fact that it ignores objects with spread operators in them and updated the commit message. Let me know if you'd like anything else!
